R/ResultFlagsIndependent.R

```diff
@@ -1131,30 +1131,80 @@ InvalidCoordinates <- function(.data,
 #' @export
 #' 
 
-identifyPotentialDuplicates <- function(.data, dist_buffer = 100){
-  dat = .data
-  dups = dat%>%dplyr::filter(!is.na(ResultMeasureValue))%>%dplyr::mutate(roundRV = round(ResultMeasureValue,digits=2))%>%dplyr::group_by(ActivityStartDate, ActivityStartTime.Time, CharacteristicName,ResultMeasureValue)%>%dplyr::summarise(numorgs = length(unique(OrganizationIdentifier)))%>%dplyr::filter(numorgs>1)
-  dups$dup_id = seq(1:dim(dups)[1])
-  
-  tdups = dplyr::left_join(dups, dat)
-  tdups$LatitudeMeasure = as.numeric(tdups$LatitudeMeasure)
-  tdups$LongitudeMeasure = as.numeric(tdups$LongitudeMeasure)
-  
-  distances = tdups%>%dplyr::ungroup()%>%dplyr::select(dup_id,LatitudeMeasure,LongitudeMeasure)
-  dcoords = sf::st_as_sf(x = distances, coords = c("LongitudeMeasure","LatitudeMeasure"), crs="EPSG:4326")
-  
-  dists = data.frame()
-  for(i in 1:max(dcoords$dup_id)){
-    ds = subset(dcoords, dcoords$dup_id==i)
-    dist = as.numeric(sf::st_distance(ds$geometry[1],ds$geometry[2]))
-    dsdist = data.frame(dup_id = i, distance_m = as.numeric(dist))
-    dsdist$TADA.idPotentialDuplicates.Flag = ifelse(dist<=dist_buffer,"POTENTIAL DUPLICATE DATAPOINT",NA)
-    dists = rbind(dists, dsdist)
+idPotentialDuplicates <- function(.data, dist_buffer = 100){
```

Comment:
Would it be difficult to add a function input here to look: 1) only within an organization for potential duplicates, or 2) across all organizations? Alternatively, not restrict this functionality to only look across organizations, but also within an org? There are scenarios where the same organization will upload the same historical data with a different site name/slightly different lat/longs as well (legacy vs current sites)

Reply:
I'd have to think through this a little, but probably. They could use TADA_NearbySites to find historic/current monitoring locations and see if duplicated data submitted across those. I noticed that exact situation in a test I did yesterday with TribalData.

Reply:
Are you thinking that the within org test would loop through each org and check all the other columns + nearby sites? That wouldn't be too bad, I don't think.

Reply:
That's good you found an example dataset with this issue for testing/dev. I think SC may be another good test state for this issue. That approach sounds like it would work